Add config variable for simple product price to default.json

zhongruige opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR adds a new config variable into `tests/e2e/env/config/default.json` which allows setting the prince for the Simple Product created in the tests. Additionally, any tests that originally hard-coded `9.99` as the default price for the Simple Product have been updated to pull in the value from the config.

Closes https://github.com/woocommerce/woocommerce/issues/28574.

### How to test the changes in this Pull Request:

1. Make sure the e2e test suite passes locally and on Travis. The tests can be ran following [these instructions](https://github.com/woocommerce/woocommerce/tree/master/tests/e2e#running-tests). For example, the `Checkout page` tests (which now use the new variable) should pass as before:
![Screen Shot 2020-12-12 at 3 53 02 PM](https://user-images.githubusercontent.com/71906536/101998623-2b0b8880-3c92-11eb-92fa-239a60154f5d.png)
2. You can update the new price configure variable to another value and the tests still pass.
3. Test backwards compatibility: copy the `config.json` file over and remove the simple product price field. After doing so, run the tests. Make sure the tests still pass.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

N/A

zhongruige committed:
Add config variable for simple product price and update tests to utilize it

rrennick commented on tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js:
```diff
@@ -19,6 +19,7 @@ const {
 const config = require( 'config' );
 
 const simpleProductName = config.get( 'products.simple.name' );
+const simpleProductPrice = config.get( 'products.simple.price' );
```
@zhongruige Any projects using the existing packages with a local `default.json` will not have the price entry in their config. To make this PR backward compatible use a default price of `9.99` when it's not found in the config.

Looking at the read me files, I see that we are missing a [`Test Variables`](https://github.com/woocommerce/woocommerce/tree/master/tests/e2e#test-variables) section in the core tests readme. This could be handled in a follow up PR.

zhongruige replied:
That makes sense--thanks for pointing out the backwards compatibility. I'll make that change.

zhongruige replied:
Just to follow up on this, I'll update the tests to have this check for the backwards compatibility:

```
const simpleProductPrice = config.has('products.simple.price') ? config.get('products.simple.price') : '9.99';
```

zhongruige committed:
Merge branch 'master' of https://github.com/woocommerce/woocommerce into add/e2e-simple-price-config-variable

zhongruige commented on tests/e2e/env/config/default.json:
```diff
@@ -12,7 +12,8 @@
   "products": {
     "simple": {
-      "name": "Simple product"
+      "name": "Simple product",
```
Good call, thanks for pointing that out @rrennick. I will make that change as well.

zhongruige committed:
Addressing merge conflicts

zhongruige committed:
Merge branch 'master' of https://github.com/woocommerce/woocommerce into add/e2e-simple-price-config-variable

zhongruige committed:
Make sure the tests are backwards compatible

rrennick reviewed (changes requested):
@zhongruige Nice work. This tested great for me. There are a couple more things needed for this PR

- Add an entry to `core-tests/CHANGELOG.md`
- Add a test / assertion at the beginning of apply coupon to test whether the simple product price is greater than `5.00`. That way the person running the test has a clear indicator of why the apply coupon test failed when the configured price is less than the discount.

zhongruige committed:
Code review feedback

zhongruige committed:
Cast to number for the comparison

zhongruige commented:
> @zhongruige Nice work. This tested great for me. There are a couple more things needed for this PR
> 
> * Add an entry to `core-tests/CHANGELOG.md`
> * Add a test / assertion at the beginning of apply coupon to test whether the simple product price is greater than `5.00`. That way the person running the test has a clear indicator of why the apply coupon test failed when the configured price is less than the discount.

Thanks for the feedback @rrennick ! I've added the entry to `core-tests/CHANGELOG.md` and added the check to the `beforeAll()` statement.

rrennick reviewed (approved):
@zhongruige This is a nice addition.
